status/status_test.go

```diff
@@ -202,6 +202,21 @@ func (s) TestFromErrorWrapped(t *testing.T) {
 	}
 }
 
+type customErrorNotImplementsInterface struct {
```

Comment:
Sorry; I believe this is 100% equivalent to `TestFromErrorUnknownError`.


Looking back at my suggestion, I believe it is nonsensical.  I was thinking more like:

```go
type grpcStatus interface {
	GRPCStatus() *grpcstatus.Status
}

type customErrorIsNilImpl struct {
	grpcStatus // embed; don't implement, so the function is `nil`
}
```

But in this case, even this type assertion in the code should succeed _and_ be non-nil:

```go
	if gs, ok := err.(grpcstatus); ok && gs != nil {
```

It will then panic when it's called, but that's fine.  We're already very deep in the "the programmer did something very wrong" territory; we don't also need to detect and prevent panics because they've fully failed to correctly implement the interface.

So I think you can just delete this test case.

Thanks!

Reply:
Wait.  Isn't this test case the same as the code in the bug report?  So does this fix the code in the bug report?

Please add a test case that fails without your changes but passes with them.  Thanks!